src/pcre2_jit_compile.c

```diff
@@ -5980,19 +6095,18 @@ while (TRUE)
         bytes = bytes_end - 32;
         }
 
-      consumed++;
-      if (--max_chars == 0)
-        return consumed;
       chars++;
       }
-    while (--repeat > 0);
+    while (--repeat > 0 && chars < chars_end);
 
+    repeat = 1;
     switch (*cc)
```

Comment:
You could merge this `switch` with the one just above. The switch statement could then set `last`, and do the `if (last) chars_end = chars` assignment at the end. Then the code to handle CRSTAR/CRQUERY/CRRANGE wouldn't be split across two switches, and it would look more similar to the OP_START/OP_QUERY/OP_RANGE code in the main switch.

Totally optional though, and not needed to merge.